### PR TITLE
removed unwrap or default and replaced with to nullable

### PR DESCRIPTION
**Pull request by NoelWidmer**
`UnwrapOrDefault()` has been removed.

`ToNullable()` existed for structs already. Now it has also been added for classes. Classes and structs are therefore more alike with this change. `ToNullable()` for classes should also cover most use cases where `UnwrapOrDefault()` was used.  A new extension class was required to work around overload issues.

If `UnwrapOrDefault()` was actually used to get the default value of a struct (which I don't think was done very often). We will now need to do: `UnwrapOr(default);`
That doesn't seem like a bad alternative at all. So I don't think we are loosing anything by removing it.

### CODE REVIEW

File: Galaxus.Functional/(Option)/OptionExtensions.cs [hunk 1]
```diff
@@ -239,15 +239,23 @@ public static async Task<T> UnwrapOrElseAsync<T>(this Task<Option<T>> self, Func
             return await fallback();
         }
 
+        #endregion
+    }
+
+    /// <summary>
+    ///     Additional extensions for the option type, providing some quality of life-shorthands.
+    /// </summary>
+    public static class OptionExtensions2
```
Comment (phkiener) [inline Galaxus.Functional/(Option)/OptionExtensions.cs:248]: Wow, TIL that type constraints are not enough to ensure two signatures are different.

I feel having `OptionExtensions2` as a name is a bit *meh*, but at the same time - the only grouping that'd make sense is putting both `ToNullable` in the same class, which again does not work. Dangit!

What about `ReferenceTypeOptionExtensions`? Most people won't see the name of the static class, but it is possible to call it like that.

**Review by phkiener — APPROVED** [review]
I'll approve the PR even with that one comment, since both is fine by me.

Wanna release this directly as 4.0.0 and push the async-extraction to 5.0.0?

File: Galaxus.Functional/(Option)/OptionExtensions.cs (hunk 1 above)
Comment (NoelWidmer) [reply Galaxus.Functional/(Option)/OptionExtensions.cs:248]: I'll put the other `.ToNullable()` in a third class called `ValueTypeOptionExtensions`. Then it is at least symetric. What do you think about that?

Comment (phkiener) [reply Galaxus.Functional/(Option)/OptionExtensions.cs:248]: Ooooh I love that! Great idea!

Comment (NoelWidmer) [reply Galaxus.Functional/(Option)/OptionExtensions.cs]: done

**Review by phkiener — APPROVED** [review]

**Pull request merged**